### PR TITLE
Geomap: Add color scale in legend

### LINKED ISSUE #47729: Geomap: Add color scale in legend

### PR DESCRIPTION
**Pull request by adela-almasan**
https://user-images.githubusercontent.com/88068998/163481797-693960c2-74d1-4c5f-aaa1-a0f10b9136f6.mov

- Reused `ColorScale` component in Geomap's legend
- Added the layer name, in addition to the field
- Added `mouseEvents` property to `MapLayerState`
- Style to be updated after UX feedback

Fixes #47729

### CODE REVIEW

File: public/app/plugins/panel/geomap/layers/data/markersLayer.tsx [hunk 1]
```diff
@@ -67,19 +67,19 @@ export const markersLayer: MapLayerRegistryItem<MarkersConfig> = {
...
-    }
-
     const style = await getStyleConfigState(config.style);
     const location = await getLocationMatchers(options.location);
     const source = new FrameVectorSource(location);
     const vectorLayer = new VectorLayer({
       source,
     });
 
+    const legendProps = new ReplaySubject<MarkersLegendProps>(1);
```
Comment (ryantxu) [inline public/app/plugins/panel/geomap/layers/data/markersLayer.tsx:77]: IIUC, we have no change here -- remove the diff just to make it smaller

**Comment by ryantxu** [comment]
This is replacing another scale right?  Can we remove it (since it is not used anymore)?

File: public/app/plugins/panel/geomap/layers/data/markersLayer.tsx (hunk 1 above)
Comment (adela-almasan) [reply public/app/plugins/panel/geomap/layers/data/markersLayer.tsx:77]: It's because the entire block was moved, so it's seen as a change.

**Comment by adela-almasan** [comment]
> This is replacing another scale right? Can we remove it (since it is not used anymore)?

Which one are you referring to? I removed the previous one from geomap

File: public/app/plugins/panel/geomap/layers/data/markersLayer.tsx (hunk 1 above)
Comment (ryantxu) [reply public/app/plugins/panel/geomap/layers/data/markersLayer.tsx:77]: I just mean, there is no reason to move it, so the diff can be smaller and only contain the one line that changed (for a reason)

This does not really matter, but when reviewing PRs it is nice to have the green/red lines all mean something :)

**Comment by ryantxu** [comment]
> Which one are you referring to? I removed the previous one from geomap

My mistake -- I see it was implemented inline, I thought it used an external component that we should also remove.  But I am wrong :)

**Review by ryantxu — APPROVED** [review]
Looks great

**Pull request merged**